File: 009-PalindromeNumber/PalindromeNumber.js
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function (x) {
  if (x < 0) return false;
  if (x == 0) return true;
  let reversed = 0;
  let original = x;
  while (x > 0) {
    reversed = reversed * 10 + (x % 10);
    x = parseInt(x / 10);
  }
  return reversed === original;
};

console.log(isPalindrome(222));

File: 0009-PalindromeNumber/PalindromeNumber.java
public class PalindromeNumber {
     public boolean isPalindrome(int x) {
        if(x < 0) return false;
        if(x == 0) return true;
        int original = x, reveresed = 0;
        while(x > 0){
            reveresed = reveresed * 10 + x % 10;
            x /= 10;
        }
        return original == reveresed;
    }
}
